### ISSUE TITLE
No video links found

### ISSUE THREAD

**Issue by grigoriynikolaev** [justchokingaround/jerry#13]
Hello.
Here is my example command and output:(script ver: 1.2.2)

> jerry -i sora no method -n 1
> Disclaimer: you need to finish the first episode before you can update your progress
> No video links found from zoro, trying gogoanime
> Watching Sora no Method - Ep: 3 The Whereabouts of Memories
> Error: sora+no+method not found

With -json argument I have this output:

> No video links found from zoro, trying gogoanime
> {"sources":[{"file":"https://tc-1.boocdn.net/_v6/62a150db2b20986d214a7867a75e002c5fbcc3e0e7ae35e63e9b43f58ac1bbac86302a21cbeb65b252731c743c8ff1557b4a8538916c5a085ad14189c552204c18241ce62e8085bd6e23adf6337da5df54691a45e0189ffc796e3dc97319799e7a0bce350861dc41a284768e16ed72e54430f773c96f798170ae4a037a47b410/master.m3u8","type":"hls"}],"sourcesBackup":[],"tracks":[{"file":"https://cc.zorores.com/81/ae/81ae2109e2877d3b3edff74d9c5ac5fe/81ae2109e2877d3b3edff74d9c5ac5fe.vtt","label":"English","kind":"captions","default":true},{"file":null,"kind":"thumbnails"}],"encrypted":false,"intro":{"start":59,"end":147},"outro":{"start":0,"end":0},"server":4}

It broke occasionally in the past with the same error, but got back after a few hours or a day. Now it's been broken for several day. I tried to watch directly in firefox and it works fine. I also tried different titles and providers, but non of them works. I get to the title selection menu, and even the name of episode I supposed to watch, but no video links.

**Comment by justchokingaround** [comment]
this is basically fixed in https://github.com/justchokingaround/jerry/blob/feat-manga/jerry.sh#L445-L457

the reason why it happens is bc every 35-36th minute of the hour (in my timezone, ie UTC + 1) the video links aren't encrypted. the bug is fixed in the `feat-manga` branch, but i haven't merged it into main yet. i need to do some testing for that branch, and this will be fixed when i merge it

**Comment by grigoriynikolaev** [comment]
Glad to hear that. I sure can wait for a merge.
About the issue. It probably was the problem before, but now I think it doesn't work at any minute of the hour. I tested it multiple times during the day.

**Comment by justchokingaround** [comment]
can you pls send logs so i can make sure that's the case:
```sh
sh -x $(which jerry) -i sora no method -n 1 2> idk.log
```

**Comment by grigoriynikolaev** [comment]
sure thing:
https://gist.github.com/grigoriynikolaev/0f12241e0faecd606f3494a7fe018e41

**Comment by justchokingaround** [comment]
try adding this in the config file:
```sh
consumet_base="api.haikei.xyz"
```

**Comment by grigoriynikolaev** [comment]
I tried. Now I haven't got even episode's name:

>  jerry -i sora no method -n 3
> Disclaimer: you need to finish the first episode before you can update your progress
> No video links found from zoro, trying gogoanime
> Error: no links found for Sora no Method episode 3/13

**Comment by justchokingaround** [comment]
can u try out the feat-manga branch pls?

**Comment by justchokingaround** [comment]
https://github.com/justchokingaround/jerry/commit/6a08fb477cf95d81e7b25aea22c6c12ab84d3d99

should fix it, try again after updating jerry with `jerry -U`

**Comment by grigoriynikolaev** [comment]
Tried. Output doesn't change. Logs:
https://gist.github.com/grigoriynikolaev/50986732c951782ae8304952a04f2de1

**Comment by justchokingaround** [comment]
oh looks like i did an oopsie doopsie

**Comment by justchokingaround** [comment]
https://github.com/justchokingaround/jerry/commit/2685dc7a97e8de5362eb7a8ee3ea579400bd7f2f

should fix it (v 1.2.4). sorry about that :skull:

**Comment by grigoriynikolaev** [comment]
Now when I try to update script using capital letter as an argument(jerry -U), it responds as if I running script to search for title named "-U".
jerry -u works fine.

After updating it's finally started working... every second time(at best). It returns same error as before, but sometimes it just works ok. I tried different titles/episodes. Doesn't seem to depend on that. I save logs from one lucky try and unlucky one:
https://gist.github.com/grigoriynikolaev/5eab9a63c51a7d77337c558b1d083d49
https://gist.github.com/grigoriynikolaev/2ff6899781b705e37b0b80f81d348b9d

Hope it'll help you somehow. It seems like its behavior now depends on that time issue you've mentioned earlier. Btw, that's how it was working for me before.

**Comment by justchokingaround** [comment]
whoops, forgot the arguments of my own script :joy: :skull: 

thx for the logs, i'll check it out as soon as i have time, a little bit busy these days, but will have more free time in like a week. i'll check this tomorrow :+1:

**Comment by RedStealthDev** [comment]
I found a way for it to work for me by disabling discord presence after i did that it has started showing me the anime and seems to work just fine.

**Comment by justchokingaround** [comment]
thanks for reminding me to fix that actually.

status update: i have some work i need to finish today, but my exams are done and i'll have some free time this week to work on jerry, hopefully i merge v4 within a few days, if all goes well, sorry for the waiting time, will be more consistent with maintaining from now on

**Comment by silvamateus** [comment]
I updated the script to v1.2.4 and started have the message
```Disclaimer: you need to finish the first episode before you can update your progress
No video links found from zoro, trying gogoanime
Error: no links found for Sora no Method episode 1/13
```

tried the feat-manga and had the same result, was about to open an issue but saw this one.

any anime, from my list or not, give the same error message.

**Comment by justchokingaround** [comment]
TLDR: Sora no Method and some other animes will work, for other animes that don't work, i'm working on a fix

To summarize, there are two bugs atm that cause this behavior:

1. The consumet api doesn't properly return an episode array for some shows. Example:
```sh
curl -s "https://api.consumet.org/meta/anilist/info/116589?provider=zoro" | tr "{|}" "\n" | tail
```
![image](https://user-images.githubusercontent.com/44473782/233185044-39545bbf-8739-4587-96fb-d4619562e95f.png)

2. There was a change in the name of one of the query parameters for the embeds, this is the case of Sora no Method.

What i will do to fix this:

1. I am considering using a recursive search function for zoro that i would implement myself, and checking the anilist id for each anime after querying the anilist title, until i find the page for which the anilist id matches. This might be expensive in terms of requests, if the title doesn't match.
2. I fixed it in `feat-manga`

**Comment by justchokingaround** [comment]
@grigoriynikolaev 
this is fixed now, as i released v1.9.0 of the script (basically a beta version for v2.0.0), the default provider is now 9anime, as it seems to be the most stable. i have not yet documented everything properly, but the `-h` and `--help` argument should give you an idea of what the script can do atm. everything is functional from my testing. sorry for the delay :pensive:
